public/index.html

```diff
@@ -1,189 +1,260 @@
-<div class="form-group" data-bs-toggle="tooltip" title="Enables finite state machine processing in Yosys. This corresponds to the 'fsm' and 'fsm -nomap' Yosys commands.">
-<label for="fsm">FSM transform (experimental)</label>
-<select id="fsm" class="form-control">
-<option value="">No FSM transform</option>
-<option value="yes">FSM transform</option>
-<option value="nomap">FSM as circuit element</option>
-</select>
-</div>
-<div class="form-group form-check" data-bs-toggle="tooltip" title="This corresponds to the 'fsm_expand' Yosys command.">
-<input type="checkbox" id="fsmexpand" class="form-check-input">
-<label for="fsmexpand" class="form-check-label">Merge more logic into FSM</label>
-</div>
-</form>
-</div>
-<div role="tabpanel" class="tab-pane tab-padded" id="iopanel">
-</div>
-</div>
-<div id="synthesize-bar">
-<form>
-<button type="submit" class="btn btn-primary">Synthesize and simulate!</button>
-</form>
-</div>
-</div>
-<div id="gutter_horiz" class="gutter gutter-horizontal"></div>
-<div id="paper">
-</div>
-<div id="gutter_vert" class="gutter gutter-vertical"></div>
-<div id="monitorbox">
-<div class="btn-toolbar" role="toolbar" aria-label="Toolbar">
-<div class="symbola btn-group mr-2" role="group" aria-label="Scale control">
-<button name="ppt_up" type="button" class="btn btn-secondary" title="Increase pixels per tick" disabled="true">+</button>
-<button name="ppt_down" type="button" class="btn btn-secondary" title="Decrease pixels per tick" disabled="true">-</button>
-</div>
-<div class="input-group mr-2">
-<div class="input-group-prepend">
-<span class="input-group-text" title="Ticks per grid line">scale</span>
-</div>
-<input type="text" class="form-control" disabled="disabled" name="scale" />
-</div>
-<div class="symbola btn-group mr-2" role="group" aria-label="Time control">
-<button name="live" type="button" class="btn btn-secondary" title="Live mode" disabled="true">▶</button>
-<button name="left" type="button" class="btn btn-secondary" title="Move left" disabled="true">←</button>
-<button name="right" type="button" class="btn btn-secondary" title="Move right" disabled="true">→</button>
-</div>
-<div class="input-group mr-2">
-<div class="input-group-prepend">
-<span class="input-group-text" title="Display range">range</span>
-</div>
-<input type="text" class="form-control" disabled="disabled" name="rangel" />
-<div class="input-group-prepend input-group-append">
-<span class="input-group-text">–</span>
-</div>
-<input type="text" class="form-control" disabled="disabled" name="rangeh" />
-</div>
-</div>
-<div id="monitor">
-</div>
-</div>
-</div>
+
+  <div id="tab-buttons-wrapper">
+    <button id="editor-tab-btn" class="tab-btn">
+      <strong style="font-size: 1.2em">&lt;/&gt;</strong><br>
+      <span style="line-height: 90%">Code</span>
+    </button>
+    <button id="circuit-tab-btn" class="tab-btn">
+      <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
```

Comment:
Why is it hard to open it in an external tool? Inline SVG is just a text so it is easy to open in any text editor.

Reply:
I don't understand what you're saying. To open this image in e.g. Inkscape, you would have to:

1. Open a text editor,
2. Cut and paste this to another file,
3. Open in Inkscape,
4. If changed, cut and paste here again.

Am I wrong?

Reply:
It's bad style. Requirements change over time. This is not a "write and forget" application, it is supposed to be maintained over the years, and this makes maintaining it harder.

By the way, what is the source of this image? Did you draw it yourself? Or did you use some resource - and in that case, is the license respected?

Reply:
Okay. If it is important for the code style, then I am going to rewrite it.

The source of the image is [svgrepo.com](https://www.svgrepo.com/svg/159891/circuit) and since its on CC0 license it does not require crediting the author, nor the origin of it.

Reply:
Okay, so I just pushed a commit that uses webpack raw loading to not inline this SVG in HTML. Please take a look and let me know if the solution is good and what should be changed if not…

Reply:
Hey @tilk, could you please give some feedback about the proposed solution with webpack raw loading?